pkg/synopsysctl/blackduck_migrate.go

```diff
@@ -0,0 +1,370 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func migrate(bd *v1.Blackduck, operatorNamespace string, flags *pflag.FlagSet) error {
+	// TODO ensure operator is installed and running a recent version that doesn't require additional migration
+
+	log.Info("Stopping Operator")
+	soOperatorDeploy, err := util.GetDeployment(kubeClient, operatorNamespace, "synopsys-operator")
+	if err != nil {
+		return err
+	}
+
+	soOperatorDeploy, err = util.PatchDeploymentForReplicas(kubeClient, soOperatorDeploy, util.IntToInt32(0))
+	if err != nil {
+		return err
+	}
+
+	// Generate Helm configuration
+	helmValuesMap, err := BlackduckV1ToHelm(bd, operatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	helmValuesMapFromFlag, err := updateBlackDuckCobraHelper.GenerateHelmFlagsFromCobraFlags(flags)
+	if err != nil {
+		return err
+	}
+
+	if err := mergo.Merge(&helmValuesMap, helmValuesMapFromFlag, mergo.WithOverride); err != nil {
+		return err
+	}
+
+	log.Info("Deleting Blackduck resources")
+	// TODO wait for resources to be deleted
+	if err := deleteComponents(bd); err != nil {
+		return err
+	}
+
+	log.Info("Upgrading Blackduck using Helm")
+	// Update the Helm Chart Location
+
+	chartLocationFlag := flags.Lookup("chart-location-path")
+	if chartLocationFlag.Changed {
+		blackduckChartRepository = chartLocationFlag.Value.String()
+	} else {
+		versionFlag := flags.Lookup("version")
+		if versionFlag.Changed {
+			blackduckChartRepository = fmt.Sprintf("%s/charts/blackduck-%s.tgz", baseChartRepository, versionFlag.Value.String())
+		}
+	}
+
+	err = util.CreateWithHelm3(bd.Name, bd.Spec.Namespace, blackduckChartRepository, helmValuesMap, kubeConfigPath, true)
+	if err != nil {
+		return fmt.Errorf("failed to create Blackduck resources: %+v", err)
+	}
+
+	// Deploy Resources
+	err = util.CreateWithHelm3(bd.Name, bd.Spec.Namespace, blackduckChartRepository, helmValuesMap, kubeConfigPath, false)
+	if err != nil {
+		return fmt.Errorf("failed to create Blackduck resources: %+v", err)
+	}
+
+	log.Info("Removing CRD")
+	if err := util.DeleteBlackduck(blackDuckClient, bd.Name, bd.Namespace, &metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+
+	log.Info("Starting Operator")
+	if _, err := util.PatchDeploymentForReplicas(kubeClient, soOperatorDeploy, util.IntToInt32(1)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isFeatureEnabled(environs []string, featureName string, expectedValue string) bool {
+	for _, value := range environs {
+		if strings.Contains(value, featureName) {
+			values := strings.SplitN(value, ":", 2)
+			if len(values) == 2 {
+				mapValue := strings.ToLower(strings.TrimSpace(values[1]))
+				if strings.EqualFold(mapValue, expectedValue) {
+					return true
+				}
+			}
+			return false
+		}
+	}
+	return false
+}
+
+func BlackduckV1ToHelm(bd *v1.Blackduck, operatorNamespace string) (map[string]interface{}, error) {
+	helmConfig := make(map[string]interface{})
+
+	// TODO handle sizes
```

Comment:
can you bubble up the leftover TODOs in the description in this PR?  or some ticket?

Reply:
Tickets would be best! I have found that people miss it otherwise

Reply:
The size is now handled during the migration